tests/EventLoopSpec.hs

```diff
@@ -162,11 +169,11 @@ initializeRepository :: FilePath -> FilePath -> IO [Sha]
 initializeRepository originDir repoDir = do
   -- Create the directory for the origin repository, and parent directories.
   FileSystem.createDirectoryIfMissing True originDir
-  shas <- populateRepository originDir
-  -- Clone with --single-branch, to make sure that we do not have all commits
+  -- Populates the repository.
+  -- Making sure to clone at the right time so that we do not have all commits
   -- in the repo dir: when this is running for real, we won't have new commits
   -- already in the repository either. They need to be fetched.
-  _    <- callGit ["clone", "--single-branch", "--branch", "master", "file://" ++ originDir, repoDir]
+  shas <- populateRepository originDir (void $ callGit ["clone", "file://" ++ originDir, repoDir])
```

Comment:
It's been some time since I touched Hoff, so sorry if my comment does not make any sense, but what's preventing us from using `git clone --branch master --depth=x`?

Reply:
Ah, ok, I think I get it - it still would not be the same as in the app itself, right?

Reply:
Yes.  Using `--depth` implies `--single-branch` which causes the local testing repo to behave differently from the local testing repo in the app itself.  Even if not, cloning at the "right time" is perhaps safer because it is what happens in reality: the remote repo getting new commits after the clone.

```
$ man git-clone
...
...
       --depth <depth>
           Create a shallow clone with a history truncated to the specified
           number of commits. Implies --single-branch unless
           --no-single-branch is given to fetch the histories near the tips of
           all branches. If you want to clone submodules shallowly, also pass
           --shallow-submodules.
...
```

Reply:
Understood! Many thanks